cli/command/container/run.go

```diff
@@ -209,7 +205,7 @@ func runRun(dockerCli *command.DockerCli, flags *pflag.FlagSet, opts *runOptions
 		})
 	}
 
-	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, hostConfig.AutoRemove)
+	statusChan := waitExitOrRemoved(ctx, dockerCli, createResponse.ID, copts.autoRemove)
```

Comment:
This specific line change did not get into the 17.03.0-ce release: https://github.com/docker/docker/blob/v17.03.0-ce/cli/command/container/run.go#L210

While other parts of this change did. Now, because the client's `ContainerCreate()` method mutates the `hostConfig` to set `AutoRemove` to `false` if the client API version is less than 1.25, nobody ends up removing the container.